src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,158 @@
+                orders.add(extractOrderFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get order from DB with user ID = "
+                    + userId, e);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders(user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create order - " + order, e);
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                return Optional.of(extractOrderFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get order from DB with ID = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                orders.add(extractOrderFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get orders from DB", e);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String queryToDeleteProducts = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToDeleteProducts)) {
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete product from shopping cart - "
+                    + order, e);
+        }
+        return insertProducts(order);
```

Comment:
I think in this method we should update both Products and userId, not only products

Reply:
We can not change the user of order, it makes no sense